util: Add -shutdownnotify option (#23395)
bitcoin/bitcoin: doc/release-notes.md

Example item
------------

New settings
------------

- The `shutdownnotify` option is used to specify a command to execute synchronously

> before Bitcoin Core has begun its shutdown sequence. (#23395)

Reviewer: As it is not clear that this pull will be merged quickly, maybe move the 5 lines to a separate file, per the instructions in https://github.com/bitcoin/bitcoin/blob/master/doc/developer-notes.md#release-notes ?

Author: Done 👍🏻

New settings
------------

- The shutdownnotify option is used to specify a command to execute when Bitcoin

> Core has finished with its shutdown sequence. (#23395)

Reviewer 2: Needs updating

> Core has finished with its shutdown sequence. (#23395)

Author: Done 👍🏻

> - The shutdownnotify option is used to specify a command to execute synchronously
> when Bitcoin Core has begins its shutdown sequence. (#23395)

Reviewer 3: ```suggestion
- The `shutdownnotify` option is used to specify a command to execute synchronously
  before Bitcoin Core begins its shutdown sequence. (#23395)
```

Author: Done 👍🏻

> 

Reviewer 4: ```suggestion

Note: Restrict scripts and commands in the shell/system used for bitcoind to avoid running potentially unsafe scripts using `shutdownnotify`

```

Context: https://github.com/bitcoin/bitcoin/pull/23395#issuecomment-956356904
More info: https://github.com/bitcoin/bitcoin/issues/23412

Author: Would it be better to add this to the end of all `-*notify`'s help option since this *note* applies to all of them?

Reviewer 4: Maybe. Not sure. Or we can add in both places. I tried to share few alternatives but if none of them look good, we should at least document related security recommendations.

Reviewer: The release notes are not the right place to put extended documentation in any case. The RPC docs, the manpage, or the docs folder are more appropriate. Also: https://github.com/bitcoin/bitcoin/pull/23395#issuecomment-956356904

Reviewer 4: I don't mind documenting it in other places. I don't think sharing a security recommendation related to new config parameter will affect anything. It can only be helpful.

I have created an issue for other details and alternatives to solve this problem. I think I can suggest changes in release notes which are part of this pull request.

Reviewer: > It can only be helpful.

A new user downloading 29.0 won't read the release notes of 23.0 (a long EOL release at that time), so putting security documentation there won't be helpful. Also, it will bloat the release notes, making it distracting to read for users that are interested in changes. So I think wrong documentation or even right documentation in the wrong place *can* be harmful.

Reviewer 4: Its okay if we just mention the link to [doc/bitcoin-conf.md](https://github.com/bitcoin/bitcoin/blob/master/doc/bitcoin-conf.md) and gets https://github.com/bitcoin/bitcoin/pull/23850 merged which adds security recommendation about command line *notify options.

Reviewer: Resolving this discussion for now. Let us know if there is anything left to be addressed.
